funnel/templates/event.html

```diff
@@ -46,37 +46,251 @@
+      if(participantList) {
+        participantList.push(participant_id);
+      }
+      else {
+        participantList = [];
+        participantList.push(participant_id);
+      }
+      return window.Store.addToLocalstorage(queueName, participantList)
+    }
+
+    //Reads first item from queue
+    //Returns undefined when queue is empty or not defined
+    window.Queue.peek = function(queueName) {
+      var participantList = window.Store.readFromLocalstorage(queueName);
+      if(participantList && participantList.length) {
+        return participantList.shift();
+      }
+    }
+
+    //Reads and returns all items from queue
+    //Returns undefined when queue is empty or not defined
+    window.Queue.peekAllItems = function(queueName) {
+      var participantList = window.Store.readFromLocalstorage(queueName);
+      if(participantList && participantList.length) {
+        return participantList;
+      }
+    }
+
+    //Removes item from queue and returns true
+    //Returns undefined when item not present in queue
+    window.Queue.dequeue = function(queueName, participant_id) {
+      var participantList = window.Store.readFromLocalstorage(queueName);
+      var index = participantList ? participantList.indexOf(participant_id) : -1;
+      if(index !== -1) {
+        //Remove item from queue and add updated queue to localStorage
+        participantList.splice(index, 1);
+        window.Store.addToLocalstorage(queueName, participantList);
+        return true;
+      }
+    }
+  }
+
   $(document).ready(function() {
+    // To disable debug mode
+    Ractive.DEBUG = false;
+
+    var participantTableContent = new Ractive({
+      el: '#participants-table-content',
+      template: '#participant-row',
+      data: {
+        participants: ''
+      }
+    });
+
+    //updateQueue: If participant in "checkin-queue" has already been checked-in then it is removed from checkin queue
+    function updateQueue(queueName) {
+      var participantIDs = window.Queue.peekAllItems(queueName);
+      if(participantIDs) {
+        participantIDs.forEach(function(participantID) {
+          var participant = "[data-pid=" + participantID +"]";
+          if(queueName === "checkin-queue") {
+            if($(participant).hasClass('js-cancel-checkin')) {
+              //Participant has been checked-in so remove from 'checkin-queue' 
+              window.Queue.dequeue("checkin-queue", participantID);
+            }
+            else {
+              $(participant).siblings('.loader').show();
+            }
+          }
+          else {
+            if($(participant).hasClass('js-checkin')) {
+              //Participant's check-in has been cancelled so remove from 'cancelcheckin-queue' 
+              window.Queue.dequeue("cancelcheckin-queue", participantID);
+            }
+            else {
+              $(participant).siblings('.loader').show();
+            }
+          }
+        });
+      }
+    }
+
+    function getParticipantsData() {
```

Comment:
This probably needs a better name, since there are multiple things happening here - getting data, setting the `Ractive` variable, and updating queues. `updateParticipantList` perhaps?